pulsar-client-cpp/tests/ServiceURITest.cc

```diff
@@ -0,0 +1,75 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include <gtest/gtest.h>
+#include "lib/ServiceURI.h"
+
+using namespace pulsar;
+
+static void verifyServiceURIFailure(const std::string& uriString, const std::string& errorMsg) {
+    try {
+        ServiceURI uri{uriString};
+        std::cerr << uriString << " should be invalid" << std::endl;
+        FAIL();
+    } catch (const std::invalid_argument& e) {
+        EXPECT_EQ(errorMsg, e.what());
+    }
+}
+
+static void verifyServiceURI(const std::string& uriString, PulsarScheme expectedScheme,
+                             const std::vector<std::string>& expectedServiceHosts) {
+    ServiceURI uri{uriString};
+    EXPECT_EQ(uri.getScheme(), expectedScheme);
+    EXPECT_EQ(uri.getServiceHosts(), expectedServiceHosts);
+}
+
+TEST(ServiceURITest, testInvalidServiceUris) {
+    verifyServiceURIFailure("localhost:6650", "The scheme part is missing: localhost:6650");
```

Comment:
Look like we don't support IPv6 yet. Do we have a plan to support it?

Reply:
Not yet. We should make multiple brokers feature available first.